### ISSUE TITLE
Pagination HTML does not show First anchor and Previous anchor not shown until after page 2

### ISSUE THREAD

**Issue by 1in9ui5t** [jhipster/generator-jhipster#2050]
The latest JHipster version 2.21.0 introduced this bug when the page number was changed to be 0-indexed. The current issues are:
1. The first page anchor is not displayed as ng-show="links['first']" always evaluates to false.
2. The previous page anchor is not displayed when the current page is 2 (index of 1) as ng-show="links['prev']" also evaluates to false (previous page has index of 0).

I'd like to submit a PR to fix this, but I'd also like to suggest a few improvements to the pagination UI if others find them useful.

Current (buggy) UI
![current](https://cloud.githubusercontent.com/assets/928636/10140650/a7cf3594-65d6-11e5-8620-e301df549ead.png)

Suggested UI
![suggested](https://cloud.githubusercontent.com/assets/928636/10140661/ae660676-65d6-11e5-98eb-6105058446d0.png)

So the proposed changes are:
1. First/Last anchors not displayed when on first/last page
2. Prev/Next anchors not displayed when on 2nd/2nd last page, instead, First/Last anchors are displayed.
3. And finally, pagination HTML is not rendered when there is only a single page of results.

**Comment by 1in9ui5t** [comment]
I must correct myself here, the "buggy" behaviour I described above is due to a merge error on my end. The first issues are not issues.

In terms of the proposed improvements...2 is already implemented...1 and 3 can still be included if others see the benefit.

I have now uncovered two "other" pagination bugs:
A. User management HTML still uses per_page parameter instead of size expected by Pageable parameter in REST controller. So user listing is currently broken.
B. When there is no content, PaginationUtil incorrectly reports rel="last" as being page -1, instead of 0.
